### DISCUSSION TITLE
RustDesk block part of the Apple icon on macos

### DISCUSSION THREAD

**Discussion by hungchieu** [rustdesk/rustdesk#8569] [Q&A]
Hi,
Is anyone notice if you let RustDesk run as a system service (for startup), it will have a small round circle, nearly transparent that block the "Apple" icon?
Attached pic.
My system is: MacOS 14.5 , RustDesk 1.2.6.

![Screenshot 2024-07-02 at 4 30 15 PM](https://github.com/rustdesk/rustdesk/assets/72768451/4cd93785-1e7e-435a-ac38-4227fb4f7a76)

**Comment by rustdesk** [comment]
What condition you can see this? I have attempted many time on my 14.5, failed to reproduce.

**Reply by rustdesk** [reply]
Login screen?

**Comment by hungchieu** [comment]
Thanks for answering.
Normal working condition, black wallpaper/background. It is easier to see if you press on the "Apple" icon. Have to press on the right side, as RustDesk icon blocked the left section.
I'm pretty sure it is from RustDesk because the that black nearly transparent icon is similar to RustDesk, and if I stop RustDesk service, it will be gone.

![Screenshot 2024-07-02 at 5 14 23 PM](https://github.com/rustdesk/rustdesk/assets/72768451/9bde54c8-3a8c-47f1-bd8e-631c17e05096)
![Screenshot 2024-07-02 at 5 15 10 PM](https://github.com/rustdesk/rustdesk/assets/72768451/83a46a3c-6685-4641-811e-2b7554e02c47)

**Comment by hungchieu** [comment]
and with proper wallpaper.
![Screenshot 2024-07-02 at 5 18 22 PM](https://github.com/rustdesk/rustdesk/assets/72768451/8149a352-9c21-4667-be36-9f057459fdd4)

**Comment by rustdesk** [comment]
Could you help stop service, then run `/Applications/RustDesk.app/Contents/MacOS/RustDesk --tray` from command line and send me the snapshot?

**Reply by hungchieu** [reply]
Oh this is correct. As soon as I run that command, the icon appears. Icon gone if stop the command.
![Screenshot 2024-07-02 at 5 28 04 PM](https://github.com/rustdesk/rustdesk/assets/72768451/34e3c71d-e439-473a-9c96-4bbbc1f98307)

**Reply by rustdesk** [reply]
It is weird, the tray icon located on your left side. On my mac, it locates on the right place.

**Reply by rustdesk** [reply]
[tao.zip](https://github.com/user-attachments/files/16064024/tao.zip)

Could you help run this?

**Comment by hungchieu** [comment]
Hi,
With/without RustDesk service, it says bad CPU type.
My system: 2.4 GHz 8-Core Intel Core i9.
![Screenshot 2024-07-02 at 5 44 44 PM](https://github.com/rustdesk/rustdesk/assets/72768451/bb4c055c-9091-47b1-a162-6298feb3e254)

**Reply by rustdesk** [reply]
Oh, you are running intel

**Reply by rustdesk** [reply]
[tao.zip](https://github.com/user-attachments/files/16064719/tao.zip)
 Please try this one, this is intel.

**Reply by hungchieu** [reply]
Hi,
Nope. It doesn't help.
Without/With RustDesk running are the same result.
![Screenshot 2024-07-02 at 9 48 03 PM](https://github.com/rustdesk/rustdesk/assets/72768451/2306be35-7a10-434a-b300-8d1ed0bf18ff)

**Reply by rustdesk** [reply]
Is it possible you install rust (https://rustup.rs/) and git (`brew install git`) on your mac? 
then 
```
git clone https://github.com/tauri-apps/tray-icon
cd tray-icon
cargo run --example tao
```

**Reply by hungchieu** [reply]
Sure. Please give me a bit of time. Those on macos are very messy. Installed rustup and missing cargo install cargo and then rustc is 1.78 which is not supported, require 1.79. rustc update then show rustup 1.79 but rustc 1.78.

**Reply by hungchieu** [reply]
Same thing.
On intel mac, macos 12.6 doesn't have this weird icon problem.

**Reply by rustdesk** [reply]
Hi, could you upload the image again? it failed.

**Reply by rustdesk** [reply]
> macos 12.6

Do you mean 14.6?

**Reply by hungchieu** [reply]
Nope, 12.6 Montery no problem at all.
Here is pic of running that Tao with RustDesk ON, Service ON.
If turn off RustDesk service then no icon at all.
![Screenshot 2024-07-02 at 11 38 56 PM](https://github.com/rustdesk/rustdesk/assets/72768451/a926303b-4f20-40a5-9c64-306e7f04b748)

**Reply by rustdesk** [reply]
This is not tao, this is still RustDesk's icon. Tao's icon is like this, 
![image](https://github.com/rustdesk/rustdesk/assets/71636191/cbc8bfef-5762-46ad-9c72-777c028c556b)

**Reply by hungchieu** [reply]
if rustdesk is not running, tao doesn't have icon.

**Reply by rustdesk** [reply]
Ok, Thanks, I got all now. We will try to get a intel mac osx 14.

**Reply by rustdesk** [reply]
We installed intel osx 14 on VM, still can not reproduce, :(

**Reply by hungchieu** [reply]
do u have log mode or verbose mode?

**Reply by rustdesk** [reply]
nothing to log or verbose for such thing.

**Reply by rustdesk** [reply]
Could you `cargo run` RustDesk?

**Reply by rustdesk** [reply]
> if rustdesk is not running, tao doesn't have icon.

Are you sure you can not see an icon like this? We use tray-icon/tao crate to show tray.

![image](https://github.com/rustdesk/rustdesk/assets/71636191/fb98a7d1-012b-42fe-9cad-82af71277b6d)

**Reply by rustdesk** [reply]
If you can not see the tao tray icon, could you please create an issue here, https://github.com/tauri-apps/tray-icon/issues?

Say you can not see tray icon with `cargo run --example tao` on intel ox 14.

**Reply by hungchieu** [reply]
Ahh, you mean on the right side? I thought you say "tao" icon will replace the RustDesk icon.
It has icon. Just the RustDesk has 2 icons. Press on the one that blocked Apple icon will display menu on the normal icon.
![Screenshot 2024-07-08 at 3 18 49 AM](https://github.com/rustdesk/rustdesk/assets/72768451/b722ce8a-32ee-4c9c-b595-d9b49a404a4a)

**Reply by rustdesk** [reply]
so we have two icons, one is black, the other is white? when you click on left side one, the menu will display on right side?

**Reply by hungchieu** [reply]
yes this is correct. The white one on the right hand side, just like all of those other icons, the black one, half transparent on the left side is blocking the Apple icon, click on either white or black will bring RustDesk menu.

**Reply by rustdesk** [reply]
Could you help run the `cargo run --example tao` again? but add below line under 
`example/tao.rs` L57
(https://github.com/tauri-apps/tray-icon/blob/d4078696edba67b0ab42cef67e6a421a0332c96f/examples/tao.rs#L57).

`.with_icon_as_template(true)`

**Reply by hungchieu** [reply]
Here it is.
![Screenshot 2024-07-08 at 4 25 22 PM](https://github.com/rustdesk/rustdesk/assets/72768451/94b1a08d-c616-40fb-8703-1aff1e2215de)
![Screenshot 2024-07-08 at 4 26 41 PM](https://github.com/rustdesk/rustdesk/assets/72768451/2dc24706-d2f3-4afc-a17d-3dbfaa7b4769)

**Reply by rustdesk** [reply]
It has only one icon? When will rustdesk show two icons? I can see only one on your new screenshot.

**Reply by rustdesk** [reply]
I found the reason https://github.com/tauri-apps/tray-icon/issues/90

**Reply by hungchieu** [reply]
Tao has 1 icon. RustDesk still shows 2 icons on my screen shoot. On the left hand side of the Apple icon, it blocked a little bit.
Yes I think it's the same issue.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/commit/c02b4f994afd27be3cd178b641486239614ce682 This fix follows the example, but there is still another open issue https://github.com/tauri-apps/tauri/issues/9480 on tao, maybe there is still some issue on the other scenario or os version.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/releases/tag/nightly ready now, please test.

**Reply by hungchieu** [reply]
yesss it is fixed! You're a legend. Thankssss.

**Reply by sweetagar** [reply]
Did I downloaded the right ver? The problem is still there... after installing rustdesk-1.2.7-x86_64-flutter3.16.9-flutter20231119.dmg the overlay is white now compared to v1.26 (overlay is black), but still there blocking half of the Apple button.

my mac is running Sonoma 14.5, 13 inch, 2018, 2.7GHz Quad-Core i7, 16GB Ram
<img width="733" alt="image" src="https://github.com/user-attachments/assets/8a6317cc-18ff-48c0-ad48-1e6063f37430">

[rustdesk-1.2.7-x86_64-flutter3.16.9-flutter20231119.dmg](https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.7-x86_64-flutter3.16.9-flutter20231119.dmg)
